src/protocol/reveal/mod.rs

```diff
@@ -166,13 +164,20 @@ mod tests {
 
         for i in 0..10_u32 {
             let record_id = RecordId::from(i);
-            let results = join3(
-                ctx[0].ctx.clone().reveal(record_id, &share[0]),
-                ctx[1].ctx.clone().reveal(record_id, &share[1]),
-                ctx[2].ctx.clone().reveal(record_id, &share[2]),
+            let input: Fp31 = rng.gen();
+
+            let m_shares = join3v(
+                zip(v.iter(), share(input, &mut rng))
+                    .map(|(v, share)| async { v.context().upgrade(record_id, share).await }),
             )
             .await;
 
+            let results =
+                join3v(zip(v.iter(), m_shares).map(|(v, m_share)| async move {
+                    v.context().reveal(record_id, &m_share).await
+                }))
+                .await;
+
             assert_eq!(input, results[0]);
             assert_eq!(input, results[1]);
             assert_eq!(input, results[2]);
```

Comment:
There's a much better way to write tests now. Try out the new hotness:

```
let things_to_reveal = [ /* array of ten randomly generated Fp31s you want to reveal */ ];
let world = TestWorld::new(QueryId);
let result = world
  .malicious(
      things_to_reveal,
      |ctx, m_shares| async move { 
          try_join_all(m_shares.enumerate().map(|(i, m_share) async move {
              ctx.reveal(m_share).await
          })),
      }
  )
  .await;
```

Or something like that.